src/testdir/test_visual.vim

```diff
@@ -1630,4 +1630,40 @@ func Test_visual_substitute_visual()
   bwipe!
 endfunc
 
+func Test_visual_getregion()
+  new
+  call setline(1, ['one', 'two', 'three'])
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>vl"
+  call assert_equal(['on'], getregion('v', '.'))
+  call assert_equal(['on'], getregion('.', 'v'))
+  call assert_equal(['o'], getregion('v', 'v'))
+  call assert_equal(['n'], getregion('.', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>Vll"
+  call assert_equal(['one'], getregion('v', '.'))
+  call assert_equal(['one'], getregion('.', 'v'))
+  call assert_equal(['one'], getregion('v', 'v'))
+  call assert_equal(['e'], getregion('.', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>\<C-v>ll"
+  call assert_equal(['one'], getregion('v', '.'))
+  call assert_equal(['one'], getregion('.', 'v'))
+  call assert_equal(['o'], getregion('v', 'v'))
+  call assert_equal(['e'], getregion('.', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>\<C-v>jj"
+  call assert_equal(['o', 't', 't'], getregion('v', '.'))
+
+  call cursor(1, 1)
+  execute "normal!" "\<ESC>\<C-v>jj$"
+  call assert_equal(['one', 'two', 'three'], getregion('v', '.'))
+
+  bwipe!
```

Comment:
Can you add a test for using this function with multibyte characters and also if the selection ends in the middle of a multibyte character?

Reply:
I am referring to the following text in the help for getregion():

```
If selection starts or ends in the middle of multibyte character, it is not included but it's selected part is substituted with spaces.
```

You can use the following test:
```
  " Selection in starts or ends in the middle of a multibyte character
  %d _
  call setline(1, ['abcdefghijk', "\U0001f1e6\u00ab\U0001f1e7\u00ab\U0001f1e8\u00ab\U0001f1e9", '1234567890'])
  call cursor(1, 3)
  call feedkeys("\<Esc>\<C-v>ljj", 'xt')
  call assert_equal(['cd', "\u00ab ", '34'], getregion('v', '.'))
  call cursor(1, 4)
  call feedkeys("\<Esc>\<C-v>ljj", 'xt')
  call assert_equal(['de', "\U0001f1e7", '45'], getregion('v', '.'))
  call cursor(1, 5)
  call feedkeys("\<Esc>\<C-v>jj", 'xt')
  call assert_equal(['e', ' ', '5'], getregion('v', '.'))
```